src/compile/compile.ts

```diff
@@ -57,10 +55,11 @@ export function compile(inputSpec: TopLevelExtendedSpec, logger?: log.LoggerInte
 }
 
 
-function getTopLevelProperties(topLevelSpec: TopLevel<any>, config: Config) {
+function getTopLevelProperties(topLevelSpec: TopLevel<any>, config: Config, autosize: AutoSizeParams) {
   return {
+    autosize: keys(autosize).length === 1 ? autosize.type : autosize,
```

Comment:
Note that the ternary operator here is not really necessary but I keep it to minimize the spec diffs.